06week/binarySearch.js

```diff
@@ -0,0 +1,1031 @@
+  "son",
+  "song",
+  "soon",
+  "sort",
+  "sound",
+  "source",
+  "south",
+  "southern",
+  "space",
+  "speak",
+  "special",
+  "specific",
+  "speech",
+  "spend",
+  "sport",
+  "spring",
+  "staff",
+  "stage",
+  "stand",
+  "standard",
+  "star",
+  "start",
+  "state",
+  "statement",
+  "station",
+  "stay",
+  "step",
+  "still",
+  "stock",
+  "stop",
+  "store",
+  "story",
+  "strategy",
+  "street",
+  "strong",
+  "structure",
+  "student",
+  "study",
+  "stuff",
+  "style",
+  "subject",
+  "success",
+  "successful",
+  "such",
+  "suddenly",
+  "suffer",
+  "suggest",
+  "summer",
+  "support",
+  "sure",
+  "surface",
+  "system",
+  "table",
+  "take",
+  "talk",
+  "task",
+  "tax",
+  "teach",
+  "teacher",
+  "team",
+  "technology",
+  "television",
+  "tell",
+  "ten",
+  "tend",
+  "term",
+  "test",
+  "than",
+  "thank",
+  "that",
+  "the",
+  "their",
+  "them",
+  "themselves",
+  "then",
+  "theory",
+  "there",
+  "these",
+  "they",
+  "thing",
+  "think",
+  "third",
+  "this",
+  "those",
+  "though",
+  "thought",
+  "thousand",
+  "threat",
+  "three",
+  "through",
+  "throughout",
+  "throw",
+  "thus",
+  "time",
+  "to",
+  "today",
+  "together",
+  "tonight",
+  "too",
+  "top",
+  "total",
+  "tough",
+  "toward",
+  "town",
+  "trade",
+  "traditional",
+  "training",
+  "travel",
+  "treat",
+  "treatment",
+  "tree",
+  "trial",
+  "trip",
+  "trouble",
+  "true",
+  "truth",
+  "try",
+  "turn",
+  "TV",
+  "two",
+  "type",
+  "under",
+  "understand",
+  "unit",
+  "until",
+  "up",
+  "upon",
+  "us",
+  "use",
+  "usually",
+  "value",
+  "various",
+  "very",
+  "victim",
+  "view",
+  "violence",
+  "visit",
+  "voice",
+  "vote",
+  "wait",
+  "walk",
+  "wall",
+  "want",
+  "war",
+  "watch",
+  "water",
+  "way",
+  "we",
+  "weapon",
+  "wear",
+  "week",
+  "weight",
+  "well",
+  "west",
+  "western",
+  "what",
+  "whatever",
+  "when",
+  "where",
+  "whether",
+  "which",
+  "while",
+  "white",
+  "who",
+  "whole",
+  "whom",
+  "whose",
+  "why",
+  "wide",
+  "wife",
+  "will",
+  "win",
+  "wind",
+  "window",
+  "wish",
+  "with",
+  "within",
+  "without",
+  "woman",
+  "wonder",
+  "word",
+  "work",
+  "worker",
+  "world",
+  "worry",
+  "would",
+  "write",
+  "writer",
+  "wrong",
+  "yard",
+  "yeah",
+  "year",
+  "yes",
+  "yet",
+  "you",
+  "young",
+  "your",
+  "yourself"
+];
+
+// my naive implementation.
+// function find(haystack, needle) {
+//   for (var i = 0; i < haystack.length; i++) {
+//     var currentWord = haystack[i];
+//     if (currentWord == needle) {
+//       return i;
+//     }
+//   }
+//   return -1;
+//}
+
+// implement a function that will
+// execute a binary search on the array
+// to find the needle
+function betterFind(haystack, needle, start, end) {
+  let mid = Math.floor((start + end) / 2);
+  if (haystack[mid] === needle) {
+    return `You found it! The needle is ${needle}`;
+  }
+  if (haystack[mid] > needle) {
```

Comment:
This will be problematic if the needs is not in the haystack at all.
Also usually, a find method returns the position the element was found at, or -1 if the element is not in the array.